IMPROVE - Contributing Docs #324

rramirez-dev opened the pull request:
# What it does 
Closes #324 

Added the following text to the CONTRIBUTING.md and getting-started.tutorial in the Documentation.docc files The latest version of firebase-tools requires a JDK of 11 or greater. You can download the latest version here [JDK23](https://www.oracle.com/java/technologies/downloads/#jdk23-mac)

# How I Tested
* From the main branch open the CONTRIBUTING.md file.
* Navigate to the Setup the Firebase Emulator section
  look for the following text under the 'npm install -g firebase-tools' bullet point
  The latest version of firebase-tools requires a JDK of 11 or greater. You can download the latest version here [JDK23](https://www.oracle.com/java/technologies/downloads/#jdk23-mac)

* From the main branch navigate to Basic-Car-Maintenance/Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial
* Locate the "Setting Up Firebase Local Emulator" Section  
* The text under "Install Firebase Tools for running  emulator" should read
   The latest version of firebase-tools requires a JDK of 11 or greater. You can download the     latest version here [JDK23](https://www.oracle.com/java/technologies/downloads/#jdk23-mac)




# Notes
N/A

# Screenshot
* Add a screenshot of your new feature! OR show a screen recording of it in action. (On the simulator press Cmd + R to record, and the stop button when done). **This video should be no longer than 30 seconds.**

rramirez-dev committed:
IMPROVE - Contributing Docs #324

Added the following text to the CONTRIBUTING.md and getting-started.tutorial in the Documentation.docc files
The latest version of firebase-tools requires a JDK of 11 or greater. You can download the latest version here
[JDK23](https://www.oracle.com/java/technologies/downloads/#jdk23-mac)

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial:
```diff
@@ -127,6 +127,7 @@
                 `brew install openjdk` 
             }
             @Step {
+                The latest version of firebase-tools requires a JDK of 11 or greater. You can download the latest version here [JDK23](https://www.oracle.com/java/technologies/downloads/#jdk23-mac)
```
What is the difference in using `brew install openjdk` vs what you get by going directly to download JDK23?

rramirez-dev replied:
According to the OpenJDK website, the install should install the JDK23 binaries. After some research it appears that in my case the 'brew install openjdk' failed to install the binaries so I had to do it manually for it to work. This ticket should be closed as the changes to the documentation are unnecessary. If contributors run into my issue we can refer them to the solution in this discussion: https://github.com/mikaelacaron/Basic-Car-Maintenance/discussions/317

mikaelacaron replied:
I think this is overall still a valid change, but worded differently it would make more sense
Because right now it looks like you have to install it twice, which *may not* be true for everyone, but adding an extra note of like "If you have issues related to Java not being installed you may have to install it directly without homebrew using this link ___"

but yeah removing all the `-` back to `*` does need to be changed back

rramirez-dev replied:
It seems that all the * were changed to -by VS Code (I assume) because I did not change them, and the original file has * instead of -. It also seems like VS Code added the extra spacing as well when I cut and pasted my content. So should I just cut and paste the contents of the original file from main into file in my branch to fix the * and spacing, and then make the other changes?

mikaelacaron replied:
Yes, you change all the `-` back to `*` and then commit and push (to the branch you're on, `rramirez-dev:324-improve-firebase-docs`) and the PR will update automatically

then click re-review button so I know it's ready to be reviewed again

mikaelacaron commented on CONTRIBUTING.md:
```diff
@@ -1,95 +1,111 @@
 # Contribution Guidelines
+
```
don't add a line break here

mikaelacaron commented on CONTRIBUTING.md:
```diff
@@ -1,95 +1,111 @@
 This document contains the rules and guidelines that developers are expected to follow, while contributing to this repository.
 
-* All contributions must NOT add any SwiftLint warnings or errors. There is a GitHub action setup for any PRs to `dev`, and Xcode will show any warnings/errors.
+- All contributions must NOT add any SwiftLint warnings or errors. There is a GitHub action setup for any PRs to `dev`, and Xcode will show any warnings/errors.
```
Don't change this a `*` or `-` works to create bullet points, and because everything already uses `*` use that

mikaelacaron commented on CONTRIBUTING.md:
```diff
@@ -1,95 +1,111 @@
+- All contributions must NOT add any SwiftLint warnings or errors. There is a GitHub action setup for any PRs to `dev`, and Xcode will show any warnings/errors.
 
 # About the Project
+
```
don't add a line break here

mikaelacaron commented on CONTRIBUTING.md:
```diff
@@ -1,95 +1,111 @@
 This app was created for [Hacktoberfest](https://hacktoberfest.com/), to help beginners in iOS dev contribute to open source. It is an app to help keep track of your car maintenance activites. This project uses [Firebase](https://firebase.google.com), [Firestore](https://firebase.google.com/products/firestore), and Sign in With Apple.
 
 ### Project Status
+
```
don't add a line break here

mikaelacaron commented on CONTRIBUTING.md:
```diff
@@ -1,95 +1,111 @@
 This app will be deployed on the Apple App Store, available for iOS 17.0 or later, [Mikaela Caron](https://github.com/mikaelacaron) (the maintainer) will upload it to the App Store after Hacktoberfest.
 
 # Getting Started
+
```
don't add a line break here

mikaelacaron commented on CONTRIBUTING.md:
````diff
@@ -1,95 +1,111 @@
 ```
 
-* Open Xcode
+- Open Xcode
+
+- In the `Basic-Car-Maintenance.xcconfig` file, fill in your `DEVELOPMENT_TEAM` and `PRODUCT_BUNDLE_IDENTIFIER`.
+  - You can find this by logging into the Apple Developer Portal
+  - This works with both free or paid Apple Developer accounts. Do **NOT** run this app on a real device due to issues with the Sign in With Apple capability.
````
Please go through the rest of this document, and change back all of the `-` back to `*`

mikaelacaron reviewed (changes requested):
Good start!! I have several comments that need resolved please. If you have any questions, you can comment directly on that thread, otherwise click the resolve button once you've addressed my comment. Once you fix everything, please click the re-review button

<img width="345" alt="Re-review button" src="https://github.com/user-attachments/assets/93238902-1fbb-4272-a8a2-84e034e65fc7">

Please go through the whole document, and change back all of the `-` back to `*`. It's unnecessary to change all of these, because they don't affect how the document is rendered in the end, and it makes a bunch of unnecessary changes in the diff of this PR

also note in your "how I tested" section it says to "from the main branch..." but that's not right, its from the `dev` branch, but also you don't really need to go into detail, because it's a documentation change. Just say like "changed section XYZ in whatever file

rramirez-dev committed:
Update CONTRIBUTING.md

The  issue with the * changing to - was because of VS Code. It changed all the * to - when ever I saved the doc. It also added the spacing.

rramirez-dev committed:
Update getting-started.tutorial

I provided a more descriptive message to the user about what 
to do if the openjdk install failed and they recieve the JDK error when they start the emulator. I spaced the message out to make it more readable.

rramirez-dev reviewed (commented):
I made the required changes in GitHub because VS Code was auto-formatting * into - when I saved the doc.

mikaelacaron committed:
Fix indentation

mikaelacaron committed:
Fix indentation and move step

mikaelacaron committed:
Fix indentation

mikaelacaron commented on Basic-Car-Maintenance/Documentation.docc/getting-started.tutorial:
```diff
@@ -166,6 +166,13 @@
                 If you don't see your user, delete the app from the simulator, and in the menu go to Device > Erase All Content and Settings (which resets your simulator), and try to run again
             }
+            
+            @Step {
+                If you receive the following error when you launch the emulator:  _'firebase-tools no longer supports Java version before 11. Please upgrade to Java version 11 or above to continue using the emulators'_
+
+                The openJDK install failed and you will have to install the latest JDK manually. You can download the latest version here [JDK23](https://www.oracle.com/java/technologies/downloads/#jdk23-mac) 
+            }
+            
```
I moved this, because your step is really after they've run the emulator, not before so I moved it in the logic flow of following the tutorial.

And note in DocC you only have two sections in a `@Step` just the direction and then optional note, you originally had multiple lines, see screenshot, which don't render, and create warnings in Xcode

![Screenshot 2024-10-06 at 10 30 16 AM](https://github.com/user-attachments/assets/b0a70e6b-26f6-447e-a0ed-4fa38af99c61)

![Screenshot 2024-10-06 at 10 30 20 AM](https://github.com/user-attachments/assets/569f1a6e-1b64-469c-bd39-37a3fa335b0e)

mikaelacaron commented on CONTRIBUTING.md:
```diff
@@ -73,6 +73,7 @@ We are going to set up the Firebase emulator to be able to load the data locally
 * Run the app
    * You should see your anonymous user in Authentication, and once you add new data, see it in Firestore emulator UI at: http://127.0.0.1:4000/firestore
    * If you don't see your user, delete the app from the simulator, and in the menu go to Device > Erase All Content and Settings (which resets your simulator), and try to run again
+   * If you receive the following error when you launch the emulator: _'firebase-tools no longer supports Java version before 11. Please upgrade to Java version 11 or above to continue using the emulators.'_ The openJDK install failed and you will have to install the latest JDK manually. You can download the latest version here [JDK23](https://www.oracle.com/java/technologies/downloads/#jdk23-mac ...
```
Also moved this one, and see how originally in the diff you still had quite a few changes with spacing / indentation, see screenshot, next time, you can always commit a few, push, and look at the diff to see if it's resolved (where it would show as nothing changed), vs this one shows a red space on the left compared to the right, that the spacing is off

![Screenshot 2024-10-06 at 10 28 16 AM](https://github.com/user-attachments/assets/4aa0519c-fb92-444b-93c2-f8f2697d462b)

mikaelacaron reviewed (approved):
Good work! I've moved around the step, because of the flow, see comments, and fixed some more indentation
